hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/EmbeddedSqmPathSource.java

```diff
@@ -18,6 +20,7 @@
 public class EmbeddedSqmPathSource<J>
 		extends AbstractSqmPathSource<J>
 		implements CompositeSqmPathSource<J> {
+	private boolean isGeneric;
```

Comment:
Can you make that final please and pass that through `buildIdAttribute` -> `SingularAttributeImpl` -> `SqmMappingModelHelper#resolveSqmPathSource` -> `EmbeddedSqmPathSource` instead of using a setter?